pkg/snmp/snmp.go

```diff
@@ -118,8 +118,12 @@ func NewListenerConfig() (ListenerConfig, error) {
 	// Set defaults before unmarshalling
 	snmpConfig.CollectDeviceMetadata = true
 	snmpConfig.CollectTopology = true
-	if err := coreconfig.Datadog.UnmarshalKey("snmp_listener", &snmpConfig, opt); err != nil {
-		return snmpConfig, err
+
+	errSnmpListenerConfig := coreconfig.Datadog.UnmarshalKey("snmp_listener", &snmpConfig, opt)
+	errNetworkDevicesSnmpListenerConfig := coreconfig.Datadog.UnmarshalKey("network_devices.snmp_listener", &snmpConfig, opt)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏

I'm unsure how this work if snmp_listener is correctly defined and network_devices is only partly defined. Do we have a chance of mixing them ?

I'm wondering if we should try something like:
```
var config
if hasConfig("network_devices.snmp_listener"):
  config, err = Unmarshal("network_devices.snmp_listener")
elif hasConfig("snmp_listener"):
  config, err = Unmarshal("snmp_listener")
else:
  return newError
```
or something along the previous lines.

Have you thought of an approach like this one?

Reply:
I also prefer your approach, I changed for it in [this commit](https://github.com/DataDog/datadog-agent/pull/24644/commits/d630868acf7f41a33037733a07436e75765681ad)